internal/querycoordv2/checkers/channel_checker.go

```diff
@@ -55,68 +55,19 @@ func (c *ChannelChecker) Check(ctx context.Context) []task.Task {
 
 func (c *ChannelChecker) checkReplica(ctx context.Context, replica *meta.Replica) []task.Task {
 	ret := make([]task.Task, 0)
-	targets := c.targetMgr.GetDmChannelsByCollection(replica.GetCollectionID())
-	dists := c.getChannelDist(replica)
 
-	lacks, redundancies := diffChannels(targets, dists)
+	lacks, redundancies := utils.GetDmChannelDiff(c.targetMgr, c.dist, c.meta, replica.GetCollectionID(), replica.GetID())
 	tasks := c.createChannelLoadTask(ctx, lacks, replica)
 	ret = append(ret, tasks...)
 	tasks = c.createChannelReduceTasks(ctx, redundancies, replica.GetID())
 	ret = append(ret, tasks...)
 
-	repeated := findRepeatedChannels(dists)
+	repeated := utils.FindRepeatedChannels(c.dist, c.meta, replica.GetID())
 	tasks = c.createChannelReduceTasks(ctx, repeated, replica.GetID())
 	ret = append(ret, tasks...)
 	return ret
 }
 
-func (c *ChannelChecker) getChannelDist(replica *meta.Replica) []*meta.DmChannel {
-	dists := make([]*meta.DmChannel, 0)
-	for _, nodeID := range replica.Nodes.Collect() {
-		dists = append(dists, c.dist.ChannelDistManager.GetByCollectionAndNode(replica.GetCollectionID(), nodeID)...)
-	}
-	return dists
-}
-
-func diffChannels(targets, dists []*meta.DmChannel) (lacks, redundancies []*meta.DmChannel) {
```

Comment:
Why move these codes to utils?

Reply:
checker should just get diff and generate task, and the diff define should not coupling with checker